processors/filterprocessor/filter_processor_test.go

```diff
@@ -0,0 +1,272 @@
+	assert.Equal(t, output.Payset[0].SignedTxnWithAD.SignedTxn.Txn.AssetTransferTxnFields.AssetReceiver, sampleAddr2)
+
+}
+
+// TestFilterProcessor_Init_All tests initialization of the filter processor with the "all" filter type
+func TestFilterProcessor_Init_All(t *testing.T) {
+
+	sampleAddr1 := basics.Address{1}
+	sampleAddr2 := basics.Address{2}
+	sampleAddr3 := basics.Address{3}
+
+	sampleCfgStr := `---
+filters:
+  - all:
+    - tag: SignedTxnWithAD.SignedTxn.Txn.PaymentTxnFields.Receiver
+      expression-type: regex 
+      expression: "` + sampleAddr2.String() + `"
+    - tag: SignedTxnWithAD.SignedTxn.Txn.Header.Sender
+      expression-type: exact
+      expression: "` + sampleAddr2.String() + `"
+`
+
+	fpBuilder, err := processors.ProcessorBuilderByName(implementationName)
+	assert.NoError(t, err)
+
+	fp := fpBuilder.New()
+	err = fp.Init(context.Background(), &conduit.PipelineInitProvider{}, plugins.PluginConfig(sampleCfgStr), logrus.New())
+	assert.NoError(t, err)
+
+	bd := data.BlockData{}
+	bd.Payset = append(bd.Payset,
+
+		transactions.SignedTxnInBlock{
+			SignedTxnWithAD: transactions.SignedTxnWithAD{
+				SignedTxn: transactions.SignedTxn{
+					Txn: transactions.Transaction{
+						PaymentTxnFields: transactions.PaymentTxnFields{
+							Receiver: sampleAddr1,
+						},
+					},
+				},
+			},
+		},
+		transactions.SignedTxnInBlock{
+			SignedTxnWithAD: transactions.SignedTxnWithAD{
+				SignedTxn: transactions.SignedTxn{
+					Txn: transactions.Transaction{
+						PaymentTxnFields: transactions.PaymentTxnFields{
+							Receiver: sampleAddr2,
+						},
+						Header: transactions.Header{
+							Sender: sampleAddr2,
+						},
+					},
+				},
+			},
+		},
+		transactions.SignedTxnInBlock{
+			SignedTxnWithAD: transactions.SignedTxnWithAD{
+				SignedTxn: transactions.SignedTxn{
+					Txn: transactions.Transaction{
+						PaymentTxnFields: transactions.PaymentTxnFields{
+							Receiver: sampleAddr3,
+						},
+					},
+				},
+			},
+		},
+	)
+
+	output, err := fp.Process(bd)
+	assert.NoError(t, err)
+	assert.Equal(t, len(output.Payset), 1)
+	assert.Equal(t, output.Payset[0].SignedTxnWithAD.SignedTxn.Txn.PaymentTxnFields.Receiver, sampleAddr2)
+	assert.Equal(t, output.Payset[0].SignedTxnWithAD.SignedTxn.Txn.Header.Sender, sampleAddr2)
+}
+
+// TestFilterProcessor_Init_Some tests initialization of the filter processor with the "any" filter type
+func TestFilterProcessor_Init(t *testing.T) {
+
+	sampleAddr1 := basics.Address{1}
+	sampleAddr2 := basics.Address{2}
+	sampleAddr3 := basics.Address{3}
+
+	sampleCfgStr := `---
+filters:
+  - any:
+    - tag: SignedTxnWithAD.SignedTxn.Txn.PaymentTxnFields.Receiver
+      expression-type: regex 
+      expression: "` + sampleAddr1.String() + `"
+    - tag: SignedTxnWithAD.SignedTxn.Txn.PaymentTxnFields.Receiver
+      expression-type: exact
+      expression: "` + sampleAddr2.String() + `"
```

Comment:
what happens if I duplicate one of the tags? Will I get duplicate blocks in result?

Reply:
Depends on the operator.  If it is "any" then if any of the duplicate tags expressions are true then it passes.  If it is all, then all the tag expressions must be true.  A use case for the latter would be to have a regex that specifies that an address begins and ends with some characters (although you could do that in one regex if you wanted).